### PR TITLE
DIG-1801: add switch for indexer

### PR DESCRIPTION
**Pull request by daisieh**
Added endpoints for turning the indexer on and off (and to check its status). 

The switch works by checking for the presence of the INDEXING_SWITCH_FILE: if it is absent at startup, the indexer quits; if it is present at startup but gets removed later, the next time something triggers the indexer, it will check for the file and quit if it's not there.

The indexing.sh loop takes care of the rest: if the indexing script is not running, it will respawn and the check will happen again.

You can manually test by setting the indexer status on and off and then running `GET http://candig.docker.internal:5080/genomics/htsget/v1/variants/LOCAL-multisample_1/index?force=true` to force an indexing.

I also added a few other exception catches for a weird edge case I ran into: if you try to index a DRS object that isn't something indexable, it should fail gracefully. The case I ran into was `GET http://candig.docker.internal:5080/genomics/htsget/v1/variants/multisample_1.vcf.gz/index?force=true`: before, it would throw a KeyError exception over and over again. Now it fails before being sent to the indexer.

### CODE REVIEW

File: htsget_server/indexing.sh [hunk 1]
```diff
@@ -1,4 +1,6 @@
 until python htsget_server/indexing.py; do
-    echo "Indexing crashed with exit code $?.  Respawning.." >&2
-    sleep 1
+    if [[ $? != 10 ]]; then
```
Comment (OrdiNeu) [inline htsget_server/indexing.sh]: As per DMs, the value of `$?` needs to be saved to a variable before the conditional here, or it'll always be `0` in the `echo`.

File: htsget_server/htsget_openapi.yaml [hunk 2]
```diff
@@ -23,6 +23,37 @@ paths:
...
+            responses:
+                200:
+                    description: Indexer status
+                    content:
+                        application/json:
+                            schema:
+                                type: object
+        post:
+            operationId: "htsget_operations.indexer_switch"
+            description: Get the status of the indexer
```
Comment (OrdiNeu) [inline htsget_server/htsget_openapi.yaml]:
```suggestion
            description: Set the indexer on or off
```

**Review by OrdiNeu — COMMENTED** [review]
I'm a bit confused as to the use-case for this; while switched OFF it's possible to create files in the indexing directory that will never be indexed (because the indexer only checks files on creation). So if the indexer is switched off accidentally while somebody else is ingesting, it becomes impossible to later index those files

**Comment by daisieh** [comment]
The indexer checks for backlog files when it starts up, so it'll always get those before starting to watch for new ones.

File: htsget_server/indexing.sh (hunk 1 above)
Comment (daisieh) [reply htsget_server/indexing.sh]: fixed.

**Review by OrdiNeu — APPROVED** [review]
:heavy_check_mark:

**Pull request merged**